### PR TITLE
Svc checker for et_schema.xml files

### PR DESCRIPTION
**Pull request by akaplya**
# Description (*)
This pr adds SVC for et_schema.xml files with are used for declaring export profiles in Magento

### CODE REVIEW

File: src/Analyzer/EtSchemaAnalyzer.php [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+    /**
+     * Analyze record structure
+     *
+     * @param string $moduleName
+     * @param $beforeRecord
+     * @param $afterRecord
+     * @return array
+     */
+    private function analyzeRecord(string $moduleName, $beforeRecord, $afterRecord): array
```
Comment (priyasbhat) [inline src/Analyzer/EtSchemaAnalyzer.php:225]: type missing for `$beforeRecord` and `$afterRecord`

File: src/Analyzer/EtSchemaAnalyzer.php [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+    {
+        $changes = [];
+        $commonFields = array_intersect(
+            array_keys($beforeRecord['field']),
+            array_keys($afterRecord['field'])
+        );
+        foreach ($commonFields as $fieldName) {
+            if (
+                $beforeRecord['field'][$fieldName]['type'] != $afterRecord['field'][$fieldName]['type']
```
Comment (priyasbhat) [inline src/Analyzer/EtSchemaAnalyzer.php:234]: Can we change it to strict comparison `!==` here and at other places below?

File: src/Analyzer/EtSchemaAnalyzer.php [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+    }
+
+    /**
+     * Analyze configuration changes
+     *
+     * @param Stmt|Registry $registryBefore
+     * @param Stmt|Registry $registryAfter
+     * @return Report
+     */
+    public function analyze($registryBefore, $registryAfter)
```
Comment (priyasbhat) [inline src/Analyzer/EtSchemaAnalyzer.php:333]: return type missing

File: src/Operation/EtSchema/EtSchemaOperation.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        $this->target = $target;
+        $this->code = $code;
+        $this->reason = $reason;
+        $this->level = $level;
+    }
+
+    /**
+     * @return string
+     */
+    public function getLocation()
```
Comment (priyasbhat) [inline src/Operation/EtSchema/EtSchemaOperation.php:53]: return type missing here and below for `getLine()` and `getLevel()`

**Review by priyasbhat — COMMENTED** [review]
Looks good to me, left a few minor comments.

**Review by priyasbhat — APPROVED** [review]

File: src/Operation/EtSchema/EtSchemaOperation.php (hunk 4 above)
Comment (akaplya) [reply src/Operation/EtSchema/EtSchemaOperation.php:53]: as design, a declaration inherited from the abstract declaration

File: src/Analyzer/EtSchemaAnalyzer.php (hunk 3 above)
Comment (akaplya) [reply src/Analyzer/EtSchemaAnalyzer.php:333]: as design, a declaration inherited from the abstract declaration

File: src/Analyzer/EtSchemaAnalyzer.php (hunk 2 above)
Comment (akaplya) [reply src/Analyzer/EtSchemaAnalyzer.php:234]: fixed

File: src/Analyzer/EtSchemaAnalyzer.php (hunk 1 above)
Comment (akaplya) [reply src/Analyzer/EtSchemaAnalyzer.php:225]: fixed

**Review by priyasbhat — APPROVED** [review]

**Comment by sidolov** [comment]
@magento import pr to magento-commerce/magento-semver

**Comment by magento-engcom-team** [comment]
@sidolov the pull request successfully imported.

**Pull request merged**